### PR TITLE
CORS fix

### PR DESCRIPTION
**Pull request by mstormi**
Fixes #528 

Cross-origin security fix
I think I implemented the outcome of your discussion
Ran it on my box without problems but would like you to confirm.

Signed-off-by: Markus Storm <markus.storm@gmx.net>

### CODE REVIEW

File: includes/nginx.conf [hunk 1]
```diff
@@ -17,11 +17,11 @@ server {
 #SSL   add_header                      Strict-Transport-Security "max-age=31536000; includeSubDomains";
 
     # Cross-Origin Resource Sharing.
-    #CORS add_header 'Access-Control-Allow-Origin' 'http://localhost:8080/rest';
-    add_header 'Access-Control-Allow_Credentials' 'true';
-    add_header 'Access-Control-Allow-Headers' 'Authorization,Accept,Origin,DNT,X-CustomHeader,Keep-Alive,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type,Content-Range,Range';
-    add_header 'Access-Control-Allow-Methods' 'GET,POST,OPTIONS,PUT,DELETE,PATCH';
-
+    add_header 'Access-Control-Allow-Origin' 'http://localhost:8080/rest,http://openhabianpi:8080/rest' always; # make sure that also a 400 response works
```
[the remark this answers is not in the document]
Comment (mstormi) [reply includes/nginx.conf:20]: Very good find. 
It's actually confusing many not just me. Best explanation I have found now is  https://stackoverflow.com/questions/25309318/best-method-access-control-allow-origin-multiple-origin-domains
I'll revert to David's "*". Ben didn't like it but it's an improvement over the current code.

**Comment by mstormi** [comment]
ok now?

**Comment by holgerfriedrich** [comment]
Not sure if I can give valuable input to this issue....
I have the feeling it is better not to put * in as default, because then all other websites could embed stuff from our server. Not doing it would mean that user runs into errors for certain setups (how common?).
I would appreciate is someone else could review this.

**Comment by mstormi** [comment]
There won't be anyone other than the two of us ... noone reviewed #528 that's been hanging around too long and that's why I had created this to revive this and get it off our todo list.
I understand this is foremost to fix errors for users (if I understand it right, _all_ to use new UIs such as PaperUI-NG or the OH3 works).
Embedding stuff will still require nginx user/pass so I don't see that to be a problem per se.
Everyone setting up remote access needs to take care of security for himself anyway.
So I'll merge.

**Pull request merged**